File: new-app/frontend/src/shared/auth/postAuthRoute.ts
/**
 * Post-auth home path — post_auth_route.dart authenticatedHomePath.
 * primaryBusiness = businesses.first (session.dart).
 */
import type { BusinessBrief } from "../api/authApi";

export function sessionIsStaff(businesses: BusinessBrief[]): boolean {
  if (businesses.length === 0) return false;
  return businesses[0].role.toLowerCase() === "staff";
}

export function authenticatedHomePath(businesses: BusinessBrief[]): string {
  return sessionIsStaff(businesses) ? "/staff/home" : "/home";
}

File: source-app/flutter_app/lib/core/router/post_auth_route.dart
import '../models/session.dart';

bool sessionIsStaff(Session session) =>
    session.primaryBusiness.role.toLowerCase() == 'staff';

/// Owner, admin, and manager may see purchase rates, totals, and margins.
bool sessionCanSeeFinancials(Session session) => !sessionIsStaff(session);

/// Owner / manager / platform super-admin may view the user list.
bool sessionCanManageUsers(Session session) {
  final r = session.primaryBusiness.role.toLowerCase();
  return r == 'owner' || r == 'admin' || r == 'manager' || session.isSuperAdmin;
}

/// Owner, admin, or platform super-admin may create staff logins.
bool sessionCanCreateUsers(Session session) {
  final r = session.primaryBusiness.role.toLowerCase();
  return r == 'owner' || r == 'admin' || session.isSuperAdmin;
}

bool sessionCanAdminUsers(Session session) {
  final r = session.primaryBusiness.role.toLowerCase();
  return r == 'owner' || r == 'admin' || session.isSuperAdmin;
}

/// Main tab shell after sign-in / splash (owner vs staff).
String authenticatedHomePath(Session session) =>
    sessionIsStaff(session) ? '/staff/home' : '/home';
